File: src/glureau/kdp/adapter/java/TargetInterface.java
package glureau.kdp.adapter.java;

import java.util.List;

/**
 * This is the interface that the clients expect. Therefore, the Adaptee class
 * must be adapted to this interface in order to be re-used by clients.
 */
interface TargetInterface {
	
	List<Double> getOperands();
	
	double sum();
	
	double max();
	
	/**
	 * This operation is not among Adaptee's ones
	 */
	double multiply();
}

File: src/glureau/kdp/adapter/kotlin/TargetInterface.kt
package glureau.kdp.adapter.kotlin

/**
 * This is the interface that the clients expect. Therefore, the Adaptee class
 * must be adapted to this interface in order to be re-used by clients.
 */
interface TargetInterface {

    val operands: List<Double>

    fun sum(): Double

    fun max(): Double

    /**
     * This operation is not among Adaptee's ones
     */
    fun multiply(): Double
}
